libs/components/popovers/testing/src/dropdown/harness/dropdown-harness.ts

```diff
@@ -0,0 +1,134 @@
+  public static hostSelector = 'sky-dropdown';
+
+  #documentRootLocator = this.documentRootLocatorFactory();
+
+  #getDropdownButton = this.locatorFor('.sky-dropdown-button');
+  #getOverlay = this.#documentRootLocator.locatorForOptional(SkyOverlayHarness);
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyDropdownHarness` that meets certain criteria.
+   */
+  public static with(
+    filters: SkyDropdownHarnessFilters
+  ): HarnessPredicate<SkyDropdownHarness> {
+    return SkyDropdownHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Clicks the dropdown button.
+   */
+  public async clickDropdownButton(): Promise<void> {
+    (await this.#getDropdownButton()).click();
+  }
+
+  /**
+   * Clicks out of the dropdown button. If `dismissOnBlur` property is set to false, then the dropdown menu does not close.
+   */
+  public async clickOutOfDropdownMenu(): Promise<void> {
+    (await (await this.#getOverlay())?.host())?.click();
+  }
+
+  /**
+   * Gets the aria-label value.
+   */
+  public async getAriaLabel(): Promise<string | null> {
+    return (await this.#getDropdownButton()).getAttribute('aria-label');
+  }
+
+  /**
+   * Gets the dropdown button style.
+   */
+  public async getButtonStyle(): Promise<string> {
+    const button = await this.#getDropdownButton();
+
+    if (await button.hasClass('sky-btn-primary')) {
+      return 'primary';
+    }
+
+    if (await button.hasClass('sky-btn-link')) {
+      return 'link';
+    }
+
+    return 'default';
+  }
+
+  /**
+   * Gets the dropdown button type.
+   */
+  public async getButtonType(): Promise<string> {
+    const button = await this.#getDropdownButton();
+
+    if (await button.hasClass('sky-dropdown-button-type-context-menu')) {
+      return 'context-menu';
+    }
+
+    return 'select';
+  }
+
+  /**
+   * Gets the dropdown menu component.
+   */
+  public async getDropdownMenu(): Promise<SkyDropdownMenuHarness | null> {
+    const dropdownMenuId = await this.#getAriaControls();
+
+    if (!dropdownMenuId) {
+      throw new Error(
+        'Unable to retrieve dropdown menu harness because dropdown is closed'
+      );
+    }
+
+    return this.#documentRootLocator.locatorForOptional(
+      SkyDropdownMenuHarness.with({ selector: `#${dropdownMenuId}` })
+    )();
+  }
+
+  /**
+   * Gets the hover tooltip text.
+   */
+  public async getTitle(): Promise<string | null> {
+    return (await this.#getDropdownButton()).getAttribute('title');
+  }
+
+  /**
+   * Gets whether the dropdown is disabled.
+   */
+  public async isDisabled(): Promise<boolean> {
+    const button = await this.#getDropdownButton();
+
+    return (await button.getProperty('disabled')) ? true : false;
+  }
+
+  /**
+   * Gets whether the dropdown menu is open.
+   */
+  public async isOpen(): Promise<boolean> {
+    const ariaExpanded = await (
+      await this.#getDropdownButton()
+    ).getAttribute('aria-expanded');
+
+    if (ariaExpanded?.match(/true/)) {
```

Comment:
Do we need the `match` function? 
```
if (ariaExpanded === 'true') {
```

Reply:
lmao i was stuck in thinking abt regex, fixed now!